### ISSUE TITLE
flaky TestGRPCLBStatsStreamingDropLoadBalancing

### ISSUE THREAD

**Issue by menghanl** [grpc/grpc-go#1557]
```
=== RUN   TestGRPCLBStatsStreamingDropLoadBalancing
E1003 19:45:25.267290    4292 grpclb.go:317] grpclb: failed to recv server list: rpc error: code = Unavailable desc = transport is closing
panic: close of closed channel
goroutine 1140 [running]:
panic(0x9f3d20, 0xc4200b0830)
	/home/travis/.gimme/versions/go1.7.6.linux.amd64/src/runtime/panic.go:500 +0x1ae
google.golang.org/grpc.(*grpclbBalancer).Start.func2.3(0xc4202e3380, 0x1, 0xc42010c780, 0xc420268cc0)
	/home/travis/gopath/src/google.golang.org/grpc/grpclb.go:486 +0xc9
created by google.golang.org/grpc.(*grpclbBalancer).Start.func2
	/home/travis/gopath/src/google.golang.org/grpc/grpclb.go:488 +0xada
```

**Comment by dfawley** [comment]
Should be fixed by #1558

**Comment by dfawley** [comment]
Something like this happened again in last night's Go1.6 cron build:

panic: close of closed channel
goroutine 821 [running]:
panic(0xa3f9c0, 0xc820010660)
	/home/travis/.gimme/versions/go1.6.4.linux.amd64/src/runtime/panic.go:481 +0x3ff
google.golang.org/grpc.(*grpclbBalancer).Start.func2.3(0xc8205c6a90, 0x1, 0xc820158480, 0xc8204dd0e0)
	/home/travis/gopath/src/google.golang.org/grpc/grpclb.go:486 +0xbc
created by google.golang.org/grpc.(*grpclbBalancer).Start.func2
	/home/travis/gopath/src/google.golang.org/grpc/grpclb.go:488 +0x15af
FAIL	google.golang.org/grpc/grpclb	12.427s

**Comment by dfawley** [comment]
Still will be fixed by #1558.
